database/migrations/2019_09_28_171753_config_fix.php

```diff
@@ -0,0 +1,577 @@
+	 *
+	 * @param array $values
+	 */
+	private function cleanup(array &$values)
+	{
+		function get_key($v)
+		{
+			return $v['key'];
+		}
+
+		$keys = array_map('get_key', $values);
+
+		try {
+			Configs::whereNotIn('key', $keys)->delete();
+		} catch (Exception $e) {
+			echo "Something weird happened.\n";
+		}
+	}
+
+	/**
+	 * Add potentially missing columns.
+	 */
+	private function missing_columns()
+	{
+		if (!Schema::hasColumn('configs', 'cat')) {
+			Schema::table('configs', function (Blueprint $table) {
+				$table->string('cat', 50)->after('value')->default('Config');
+			});
+		}
+		if (!Schema::hasColumn('configs', 'confidentiality')) {
+			Schema::table('configs', function (Blueprint $table) {
+				$table->tinyInteger('confidentiality')->after('cat')
+					->default(0);
+			});
+		}
+		if (!Schema::hasColumn('configs', 'type_range')) {
+			Schema::table('configs', function (Blueprint $table) {
+				$table->string('type_range')->after('cat')->default('0|1');
+				$table->string('description')->after('confidentiality')
+					->default('');
+			});
+		}
+	}
+
+	/**
+	 * Update the fields which are missing, set up the correct values for columns.
+	 *
+	 * @param array $default_values
+	 */
+	private function update_missing_fields(array &$default_values)
+	{
+		foreach ($default_values as $value) {
+			$config = Configs::updateOrCreate(['key' => $value['key']],
+				[
+					'cat' => $value['cat'],
+					'type_range' => $value['type_range'],
+					'confidentiality' => $value['confidentiality'],
+				]);
+			if ($config->value == '') {
+				$config->value = $value['value'];
+				$config->save();
+			}
+		}
+	}
+
+	/**
+	 * Run the migrations.
+	 *
+	 * @return void
+	 */
+	public function up()
+	{
+		define('INT', 'int');
+		define('STRING', 'string');
+		define('STRING_REQ', 'string_required');
+		define('BOOL', '0|1');
+		define('TERNARY', '0|1|2');
+		define('DISABLED', '');
+
+		$default_values = [
+			[
+				'key' => 'version',
+				'value' => '040000',
+				'cat' => 'Admin',
+				'type_range' => INT,
+				'confidentiality' => '0',
+			],
+			[
+				'key' => 'username',
+				'value' => '',
+				'cat' => 'Admin',
+				'type_range' => STRING_REQ,
+				'confidentiality' => '4',
+			],
+			[
+				'key' => 'password',
+				'value' => '',
+				'cat' => 'Admin',
+				'type_range' => STRING_REQ,
+				'confidentiality' => '4',
+			],
+			[
+				'key' => 'check_for_updates',
+				'value' => '0',
+				'cat' => 'Admin',
+				'type_range' => BOOL,
+				'confidentiality' => '0',
+			],
+			[
+				'key' => 'sorting_Photos_col',
+				'value' => 'title',
+				'cat' => 'Gallery',
+				'type_range' => 'id|takestamp|title|description|public|star|type',
+				'confidentiality' => '2',
+			],
+			[
+				'key' => 'sorting_Photos_order',
+				'value' => 'ASC',
+				'cat' => 'Gallery',
+				'type_range' => 'ASC|DESC',
+				'confidentiality' => '2',
+			],
+			[
+				'key' => 'sorting_Albums_col',
+				'value' => 'max_takestamp',
```

Comment:
By the look of it, this one used to default to `description`, which in retrospect I agree probably wasn't the best choice... :wink: